File: src/main/java/com/pfe/elearning/course/dto/CourseResponse.java
package com.pfe.elearning.course.dto;

import com.pfe.elearning.lesson.dto.LessonResponse;
import com.pfe.elearning.topic.dto.TopicResponse;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.List;


@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor

public class CourseResponse {
    private Integer id;
    private String title;
    private String description;
    private LocalDateTime createdAt;
  //  private String topicTitle;
    private String imageUrl;
    private String duration;
    private String publisherName; // Add this field for the publisher's name
    private TopicResponse topic;
    private List<LessonResponse> lessons;

}

File: src/app/models/course/course-response.ts
import {TopicResponse} from "../topic/topic-response";
import {LessonResponse} from "../lesson/lesson-response";

export interface CourseResponse {
  id: number;
  title: string;
  description: string;
  createdAt: Date;
  imageUrl?: string;
  duration?: string;
  publisherName: string;
  topic: TopicResponse;
  lessons: LessonResponse[];
}
